src/util.h

```diff
@@ -571,6 +571,15 @@ std::string json_escape_utf8(const std::string& val, bool escape_printable_contr
  */
 std::string json_escape_utf8(const char* val, size_t val_size, bool escape_printable_controls = true);
 
+/**
+ * Checks for values that are approximately equal.
+ * @param a first value to compare
+ * @param b second value to compare
+ * @param tolerance how close they need to be to deem them "approximately equal"
+ * @return true if `a` is within the given tolerance of `b`, false otherwise
+ */
+bool approx_equal(double a, double b, double tolerance = 1e-6);
```

Comment:
Not sure about this one, but for general `double` values the default epsilon seems pretty big. What do you think about using e.g., [`std::numeric_limits<double>::epsilon()`](https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon) as default value and update the callers to pass in context-specific bigger value where they did not specify an epsilon before?